src/include/units/physical/si/density.h

```diff
@@ -0,0 +1,47 @@
+// The MIT License (MIT)
+//
+// Copyright (c) 2018 Mateusz Pusz
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+#pragma once
+
+#include <units/physical/dimensions.h>
+#include <units/physical/si/mass.h>
+#include <units/physical/si/length.h>
+#include <units/physical/si/prefixes.h>
+#include <units/quantity.h>
+
+namespace units::si {
+
+struct kilogram_per_metre_cub : unit<kilogram_per_metre_cub> {};
+
+struct dim_density : physical::dim_density<dim_density, kilogram_per_metre_cub,dim_mass, dim_length> {};
+
+template<Unit U, Scalar Rep = double>
+using density = quantity<dim_density, U, Rep>;
+
+inline namespace literals {
+
+constexpr auto operator"" kg_per_m3(unsigned long long l) { return density<kilogram_per_metre_cub, std::int64_t>(l); }
+constexpr auto operator"" kg_per_m3(long double l) { return density<kilogram_per_metre_cub, long double>(l); }
```

Comment:
I am not sure about `_per_`. For example we have `mps` and `kmph` in velocity. Should we be consistent here?

Reply:
Another inconsistency is long vs. short unit names, e.g. `kg` vs. `kiloohm`. I know, it is pretty much bikeshedding at its ugliest, but have you considered using Unicode characters for UDL? So `kiloohm` could become `kΩ` (using ["ohm sign"](http://www.unicode-symbol.com/u/2126.html) character seems appropriate, isn't it?) and in this case `kg_per_m3` could become `kg∕m³` (using ["division slash"](http://www.unicode-symbol.com/u/2215.html) instead of common `/` and ["superscript three"](http://www.unicode-symbol.com/u/00B3.html)). Typing these using just a keyboard won't be easy, but with some help from IDE plugins it should be OK. Many science-oriented IDEs these days can automatically convert TeX notation into Unicode.

Reply:
Maybe need to implement a large set of units, then can get a better idea of what problems there may be.

Reply:
Fortress experimented with non-ascii source code... 
https://en.wikipedia.org/wiki/Fortress_(programming_language)

Reply:
Here is a good summary across various programming languages:
https://rosettacode.org/wiki/Unicode_variable_names

Reply:
> I am not sure about `_per_`. For example we have `mps` and `kmph` in velocity. Should we be consistent here?

Yes. Sorry, so these should be kgpm3 ?

Reply:
> > I am not sure about `_per_`. For example we have `mps` and `kmph` in velocity. Should we be consistent here?
> 
> Yes. Sorry, so these should be kgpm3 ?

Done : https://github.com/kwikius/units/blob/kwikius_examples_v2/src/include/units/physical/si/density.h#L42